### PR TITLE
[Dashboard] Remove non-branding icons

### PR DESCRIPTION
**Pull request by kien-ngo**
## PR-Codex overview
This PR primarily focuses on replacing icons from the `@react-icons/all-files` library with equivalent icons from the `lucide-react` library across various components, enhancing consistency and potentially improving performance.

### Detailed summary
- Removed `IconType` prop from several interfaces.
- Replaced `@react-icons/all-files` icons with `lucide-react` icons in components like `HackathonFooter`, `NFTClaimButton`, `SelectOption`, and others.
- Updated icon usage in the `MobileNavSection` and `NavCard`.
- Adjusted styling for new icons where necessary.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by kien-ngo** [comment]
* **#4810** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4810?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kien-ngo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/components/buttons/MismatchButton.tsx [hunk 1]
```diff
@@ -448,7 +447,7 @@ const MismatchNotice: React.FC<{
     <div className="flex flex-col gap-4">
       <Heading size="label.lg">
         <div className="flex flex-row items-center gap-2">
-          <Icon boxSize={6} as={AiOutlineWarning} />
+          <TriangleAlert />
```
Comment (MananTank) [inline apps/dashboard/src/components/buttons/MismatchButton.tsx:450]: size class missing

File: apps/dashboard/src/components/hackathon/HacakthonEarnFooter.tsx [hunk 2]
```diff
@@ -60,7 +60,7 @@ export const HackathonEarnFooter = ({
           h="68px"
           w={{ base: "90%", md: 80 }}
           fontSize="20px"
-          leftIcon={<Icon as={ImMagicWand} />}
+          leftIcon={<Wand />}
```
Comment (MananTank) [inline apps/dashboard/src/components/hackathon/HacakthonEarnFooter.tsx:63]: size class missing, also prefer the name with "Icon` postfix - eg: `WandIcon` here

File: apps/dashboard/src/components/hackathon/HackathonFooter.tsx [hunk 3]
```diff
@@ -56,7 +56,7 @@ export const HackathonFooter = ({
           h="68px"
           w={{ base: "90%", md: 80 }}
           fontSize="20px"
-          leftIcon={<Icon as={ImMagicWand} />}
+          leftIcon={<Wand />}
```
Comment (MananTank) [inline apps/dashboard/src/components/hackathon/HackathonFooter.tsx:59]: same thing here

File: apps/dashboard/src/components/homepage/CodeBlock/index.tsx [hunk 4]
```diff
@@ -162,10 +161,11 @@ export const HomePageCodeBlock: React.FC<CodeBlockProps> = ({
               colorScheme="gray"
               size="sm"
               icon={
-                <Icon
-                  as={hasCopied ? IoMdCheckmark : FiCopy}
-                  fill={hasCopied ? "green.500" : undefined}
-                />
+                hasCopied ? (
+                  <Check className="text-green-500" size={14} />
```
Comment (MananTank) [inline apps/dashboard/src/components/homepage/CodeBlock/index.tsx:165]: use size class instead, prefer CheckIcon

File: apps/dashboard/src/components/shared/FileInput.tsx [hunk 5]
```diff
@@ -182,14 +181,13 @@ export const FileInput: React.FC<IFileInputProps> = ({
               disabled={isDisabled}
               className="gap-2"
             >
-              <AiOutlineFileAdd className="size-4" />
-              {selectOrUpload} {helperTextOrFile}
+              <FilePlus size={16} /> {selectOrUpload} {helperTextOrFile}
```
Comment (MananTank) [inline apps/dashboard/src/components/shared/FileInput.tsx:184]: prefer FilePlusIcon name, use size class

File: apps/dashboard/src/components/shared/FileInput.tsx [hunk 6]
```diff
@@ -182,14 +181,13 @@ export const FileInput: React.FC<IFileInputProps> = ({
...
             </Button>
           )}
           {noDisplay && (
             <Link href={fileUrl} target="_blank" className="no-underline">
               <Button variant="outline" className="gap-2">
-                <AiFillEye className="size-4" />
+                <Eye size={16} />
```
Comment (MananTank) [inline apps/dashboard/src/components/shared/FileInput.tsx:190]: same - name + size class

File: apps/dashboard/src/contract-ui/tabs/embed/components/embed-setup.tsx [hunk 7]
```diff
@@ -567,7 +566,7 @@ export const EmbedSetup: React.FC<EmbedSetupProps> = ({
                 chainId,
               });
             }}
-            leftIcon={hasCopied ? <IoMdCheckmark /> : <FiCopy />}
+            leftIcon={hasCopied ? <Check size={14} /> : <Copy size={14} />}
```
Comment (MananTank) [inline apps/dashboard/src/contract-ui/tabs/embed/components/embed-setup.tsx:569]: same - name + size class

File: apps/dashboard/src/contract-ui/tabs/events/components/events-feed.tsx [hunk 8]
```diff
@@ -398,7 +398,7 @@ const TransactionData: React.FC<TransactionDataProps> = ({
             }
           >
             <div className="flex items-center justify-center">
-              <Icon as={AiOutlineQuestionCircle} color="gray.600" />
+              <CircleHelp className="text-gray-600" size={14} />
```
Comment (MananTank) [inline apps/dashboard/src/contract-ui/tabs/events/components/events-feed.tsx:401]: name + size class

File: apps/dashboard/src/contract-ui/tabs/nfts/components/claim-button.tsx [hunk 9]
```diff
@@ -28,7 +28,7 @@ export const NFTClaimButton: React.FC<NFTClaimButtonProps> = ({ contract }) => {
       </Drawer>
       <Button
         colorScheme="primary"
-        leftIcon={<Icon as={GiDiamondHard} />}
+        leftIcon={<Gem size={16} />}
```
Comment (MananTank) [inline apps/dashboard/src/contract-ui/tabs/nfts/components/claim-button.tsx:31]: name + size class

File: apps/dashboard/src/core-ui/batch-upload/batch-lazy-mint.tsx [hunk 10]
```diff
@@ -396,10 +395,11 @@ const SelectReveal: React.FC<SelectRevealProps> = ({
...
                       <InputRightElement cursor="pointer">
-                        <Icon
-                          as={show ? AiFillEye : AiFillEyeInvisible}
-                          onClick={() => setShow(!show)}
-                        />
+                        {show ? (
+                          <Eye onClick={() => setShow(!show)} size={12} />
+                        ) : (
+                          <EyeOff onClick={() => setShow(!show)} size={12} />
+                        )}
```
Comment (MananTank) [inline apps/dashboard/src/core-ui/batch-upload/batch-lazy-mint.tsx:402]: name + size class

File: apps/dashboard/src/core-ui/batch-upload/lazy-mint-form/select-option.tsx [hunk 11]
```diff
@@ -97,7 +90,7 @@ export const SelectOption: React.FC<SelectOptionProps> = ({
                 }
               >
                 <Flex alignItems="center">
-                  <Icon as={AiOutlineInfoCircle} boxSize={5} />
+                  <Info size={16} />
```
Comment (MananTank) [inline apps/dashboard/src/core-ui/batch-upload/lazy-mint-form/select-option.tsx:93]: name + size class

File: apps/dashboard/src/pages/hackathon/consumer-crypto.tsx [hunk 12]
```diff
@@ -138,7 +138,7 @@ const Hackathon = () => {
               h="68px"
               w={{ base: "100%", md: 96 }}
               fontSize="20px"
-              leftIcon={<Icon as={ImMagicWand} />}
+              leftIcon={<Wand />}
```
Comment (MananTank) [inline apps/dashboard/src/pages/hackathon/consumer-crypto.tsx:141]: name + size class

File: apps/dashboard/src/pages/hackathon/earn.tsx [hunk 13]
```diff
@@ -149,7 +149,7 @@ const HackathonEarn = () => {
               h="68px"
               w={{ base: "100%", md: 96 }}
               fontSize="20px"
-              leftIcon={<Icon as={ImMagicWand} />}
+              leftIcon={<Wand />}
```
Comment (MananTank) [inline apps/dashboard/src/pages/hackathon/earn.tsx:152]: name + size class

File: apps/dashboard/src/tw-components/code-block.tsx [hunk 14]
```diff
@@ -103,10 +101,11 @@ export const CodeBlock: React.FC<CodeBlockProps> = ({
               colorScheme="gray"
               size="sm"
               icon={
-                <Icon
-                  as={hasCopied ? IoMdCheckmark : FiCopy}
-                  fill={hasCopied ? "green.500" : undefined}
-                />
+                hasCopied ? (
```
Comment (MananTank) [inline apps/dashboard/src/tw-components/code-block.tsx:104]: name + size class

**Review by MananTank — APPROVED** [review]
thanks!

**Pull request merged**